cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -94,6 +97,17 @@ func PerformPostUpgradeTasks(client clientset.Interface, cfg *kubeadmapi.InitCon
 		errs = append(errs, err)
 	}
 
+	coreDNSConfigMap, corefile, _, err := dns.GetCoreDNSInfo(client)
+	if err != nil {
+		errs = append(errs, err)
+	}
+	isMigrationRequired := dns.IsCoreDNSConfigMapMigrationRequired(corefile)
+	if isMigrationRequired {
```

Comment:
In fact, this whole blob needs to go into `EnsureDNSAddon` or a new func called `UpgradeDNSAddon`.
Even with that in mind, I don't think, that we should backup the CoreDNS configuration into a config map. We may have it locally in a string var, just in case we want to revert to it.

Reply:
> We may have it locally in a string var, just in case we want to revert to it.

one problem with this approach is that if kubeadm exists the backup will be lost.
if it's stored on disk it is better.

Reply:
Indeed a comprehensive crash tolerant upgrade revert system is a good thing to have, but we don't have the time, plus cluster backup tools already provide this sort of functionality.